rfcs/0001-rfc-process.md

```diff
@@ -0,0 +1,135 @@
+* A core team member directly invites you to open an RFC from a community or support channel like GitHub Discussions, Slack, or Twitter
+* You open a "[RRFC](https://github.com/lit/rfcs/issues/new)" (Requesting Request For Comment) issue in lit/rfcs and get positive feedback 
+
+### Opening an RFC
+
+1. Fork the lit/rfcs repo
+2. Draft the RFC
+    1. Add the RFC file to /rfcs in a markdown file named `NNNN-_descriptive-name_.md`
+        1. Note the `NNNN` is literal. RFCs will not have an assigned RFC number when first opened.
+3. Open an RFC pull request
+    1. Optionally share the draft on [GH discussions](https://github.com/lit/lit/discussions) or [Slack](https://lit.dev/slack-invite/) before opening a pull request
+    2. Once opened, update the PR with a link to the RFC PR in the metadata. This allows readers of the RFCs to find the review discussions.
+
+### Discussion and Iteration
+
+RFCs must be discussed and reviewed by the relevant maintainers, Lit project leadership, and other stakeholders before being accepted or rejected.
+
+#### Reviewers
+
+RFCs must be reviewed by Lit project leadership and by project area maintainers. Lit project leadership who can approve RFCs are listed in the lit/rfcs repo's [CODEOWNERS](../CODEOWNERS) file. RFCs must be approved by leadership before moving to accepted stage and being merged. Area maintainers are sometimes identifiable by who is active on certain projects in the lit/lit monorepo, but the Lit team can add the appropriate people as reviews if it's unclear.
+
+#### Timeline
+
+Many RFC processes have a prescribed timeline for discussion. We aren't yet sure what timelines for feedback are both most productive and attainable. For the time being we aim to have initial feedback within about 7 days and an accepted or rejected decision within about 30 days.
+
+#### Acceptance
+
+Once accepted, RFCs are assigned an RFC number and pull requests are updated and merged. This means that all RFCs in lit/rfcs start as accepted. Rejected RFC pull requests are closed.
+
+#### RFC Meetings
+
+RFC stakeholders may decide that RFCs need or could benefit from live community discussion. They will be added to the RFC meetings agenda list and discussed at the next opportunity.
+
+### Updates and Amendments
+
+After being accepted RFCs are intended to mostly stay unchanged.
+
+Major changes proposed to an RFC must be done in a new RFC. If a new RFC deprecates a previous RFC, it must mark the previous RFC as deprecated with a link to the new RFC. Otherwise the new RFC should update the previous RFC with a link and callouts to the new RFCs in relevant places.
+
+Small changes to RFCs can be made to:
+
+* Add clarifications
+* Update metadata and other non-design text (examples, documentation)
+* Minor changes to implementation, possibly based on implementor feedback
+
+### Implementation
+
+As RFCs are implemented, they should be updated to indicate so. The status is updated to *Implemented* and relevant PRs linked.
```

Comment:
In a team discussion about how the RFC process relates to design docs (and an ideal world where the RFC is updated with most details that might otherwise go in a design doc to avoid duplication), we noted that the [Vue RFC process](https://github.com/vuejs/rfcs) has a pretty nice definition of states/status:

- **Pending**: when the RFC is submitted as a PR.
- **Active**: when an RFC PR is merged and undergoing implementation.
- **Landed**: when an RFC's proposed changes are shipped in an actual release.
- **Rejected**: when an RFC PR is closed without being merged.

With these definitions, most of the design scrutiny will take place _on the RFC_ in the "Pending" state, before the PR is merged -- with the goal of getting it as close to de-risked as possible without writing any code.

However, once merged and in the "Active" state, most of the subsequent design scrutiny will naturally shift to the actual implementation _code in PRs_, such that updates to the RFC in the "Active" state would mostly be in service of reflecting decisions that are being carried out in the process of implementing the RFC, and should in practice be be mostly rubber-stamp updates to the RFC at that point to keep it in sync with the implementation.

We'd still do PR's to both, but the idea is that once we get to implementing, if on the code PR we say "yeah, in retrospect we should make a new public API for that" and maintainers agree to that change on the code PR, we shouldn't have to do another heavyweight review process to update the "Active" RFC at that point to reflect the decision; the PR to the RFC at that point could just be "Updates based on implementation review in code PR ##" and we wouldn't expect a ton of back/forth on that.

So with that slight nuance (and maybe explicitly articulating the above states and expectations), this seems like it could plausibly replace and open up a lot of design docs, without adding a heavyweight process.

Reply:
I think this is more-or-less describing what I intend here, with the additional clarification of leeway for changes during implementation, which sounds reasonable to me.

Let me update the language and rename "Accepted" to "Active".

Reply:
PTAL